osmdroid-android/src/main/java/org/osmdroid/views/Projection.java

```diff
@@ -703,36 +722,49 @@ private void adjustOffsets(final long pDeltaX, final long pDeltaY) {
 	}
 
 	/**
+	 * @param pPixelMin Pixel position of the limit (left)
+	 * @param pPixelMax Pixel position of the limit (right)
+	 * @param pWorldSize World map size - for modulo adjustments
+	 * @param pScreenSize Screen size
+	 * @param pExtraSize Extra size to consider at each side of the screen
+	 * @return the offset to apply so that the limits are within the screen
 	 * @since 6.0.0
 	 */
-	public long getScrollX() {
-		return mScrollX;
-	}
+	public static long getScrollableOffset(final long pPixelMin, long pPixelMax,
+										   final double pWorldSize,
+										   final int pScreenSize, final int pExtraSize) {
+		while (pPixelMax - pPixelMin < 0) { // date line + several worlds fix
+			pPixelMax += pWorldSize;
+		}
 
-	/**
-	 * @since 6.0.0
-	 */
-	public long getScrollY() {
-		return mScrollY;
+		long delta;
+		if (pPixelMax - pPixelMin < pScreenSize - 2 * pExtraSize) {
+			final long half = (pPixelMax - pPixelMin) / 2;
+			if ((delta = pScreenSize / 2 - half - pPixelMin) > 0) {
+				return delta;
+			}
+			if ((delta = pScreenSize / 2 + half - pPixelMax) < 0) {
+				return delta;
+			}
+			return 0;
+		}
+		if ((delta = pExtraSize - pPixelMin) < 0) {
+			return delta;
+		}
+		if ((delta = pScreenSize - pExtraSize - pPixelMax) > 0) {
+			return delta;
+		}
+		return 0;
 	}
 
 	/**
 	 * @since 6.0.0
 	 */
-	private long checkScrollableOffset(
-			final long pPixelMin, long pPixelMax, final double pWorldSize, final int pScreenSize) {
-		while (pPixelMax - pPixelMin < 0) { // date line + several worlds fix
-			pPixelMax += pWorldSize;
-		}
-
-		if (pPixelMax - pPixelMin < pScreenSize) {
-			return pPixelMax - pScreenSize / 2 - (pPixelMax - pPixelMin) / 2;
-		} else if (pPixelMin >= 0) {
-			return pPixelMin;
-		} else if (pPixelMax <= pScreenSize) {
-			return pPixelMax - pScreenSize;
-		} else {
-			return 0;
+	boolean setMapScroll(final MapView pMapView) {
```

Comment:
should this be private or protected?

Reply:
My guess is package-private, as it's meant to be used only by `MapView`, which happens to be in the same package.